Wrappers/Python/cil/optimisation/functions/Function.py

```diff
@@ -438,7 +432,8 @@ def __call__(self, x):
         --------
         DataContainer, the value of the scaled function.
         """
-        return self.scalar * self.function(x)
+        res = self.function(x)
+        return dtype_like(self.scalar, res) * res
```

Comment:
Why is casting necessary here? AFAIK this `self.function(x)` should return a number (probably a float32) and `self.scalar` is also a number.

Reply:
I asked about `self.function` and nobody seemed to know whether it's guaranteed not to return an array... wdyt?

Reply:
I mean, this is paranoid but it will certainly work.